hs_geo_raster_resource/migrations/custom_migration_for_raster_ori_cov_update_20161107.py

```diff
@@ -0,0 +1,105 @@
+
+from django.db import migrations
+from django.core.files.uploadedfile import UploadedFile
+
+from django_irods.storage import IrodsStorage
+from hs_core import hydroshare
+from hs_core.hydroshare.utils import resource_modified, get_file_from_irods
+from hs_geo_raster_resource.models import RasterResource
+from hs_geo_raster_resource import raster_meta_extract
+
+
+def migrate_tif_file(apps, schema_editor):
+    log = logging.getLogger()
+
+    copy_res_fail = []
+    meta_update_fail = []
+    meta_update_success = []
+
+    # start migration for each raster resource that has raster files
+    for res in RasterResource.objects.all():
+        if res.files.all():
+            # copy all the resource files to temp dir
+            temp_dir = ''
+            res_file_tmp_path = ''
+
+            try:
+                temp_dir = tempfile.mkdtemp()
+                for res_file in res.files.all():
+                    res_file_tmp_path = get_file_from_irods(res_file)
+                    shutil.copy(res_file_tmp_path,
+                                os.path.join(temp_dir, os.path.basename(res_file_tmp_path)))
+                    shutil.rmtree(os.path.dirname(res_file_tmp_path))
+
+                vrt_file_path = [os.path.join(temp_dir, f) for f in os.listdir(temp_dir) if '.vrt' == f[-4:]].pop()
+
+            except Exception as e:
+                if os.path.isdir(temp_dir):
+                    shutil.rmtree(temp_dir)
+                if os.path.isfile(res_file_tmp_path):
+                    shutil.rmtree(os.path.dirname(res_file_tmp_path))
+                log.exception(e.message)
+                copy_res_fail.append('{}:{}'.format(res.short_id, res.metadata.title.value))
+                continue
+
+            # update the metadata for the original coverage information of all the raster resources
+            try:
+                meta_updated = False
+
+                # extract meta
+                ori_dir = os.getcwd()
+                os.chdir(temp_dir)
+                res_md_dict = raster_meta_extract.get_raster_meta_dict(vrt_file_path)
+                os.chdir(ori_dir)
+                shutil.rmtree(temp_dir)
+
+                # update original coverage information for datum and coordinate string in django
+                if res.metadata.originalCoverage:
+                    res.metadata.originalCoverage.delete()
+                    v = {'value': res_md_dict['spatial_coverage_info']['original_coverage_info']}
+                    res.metadata.create_element('OriginalCoverage', **v)
+                    meta_updated = True
+
+                # update the bag if meta is updated
+                if meta_updated:
+                    bag_name = 'bags/{res_id}.zip'.format(res_id=res.short_id)
+                    istorage = res.get_irods_storage()
+                    if istorage.exists(bag_name):
+                        # delete the resource bag as the old bag is not valid
+                        istorage.delete(bag_name)
```

Comment:
This is aimed to delete the existing bag to avoid the situation that after migration the bag becomes invalid . ( When the bag is downloaded, it can't be unzipped correctly sometimes.)